### PR TITLE
disable MHA_AVX2

### PR DESCRIPTION
**Pull request by yuchengliu1**
## Type of Change

disable MHA_AVX2

## Description

detail description 
Issues: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Review by DDEle — APPROVED** [review]

File: neural_speed/core/layers/mha_dense.cpp [hunk 1]
```diff
@@ -72,7 +72,9 @@ bool bestla_reordered_attn_fp32_support(const attn_shape_t* params) {
   // TODO(Yi): check K V's layout
   if (_cd->AMX_BF16()) return true;
 #endif
-  return !_cd->AVX512F() || _cd->AVX2();  // use avx2 and f16c on avx2 platforms
+  // use avx2 and f16c on avx2 platforms
+  // todo: check avx2 mha on sever
+  return !_cd->AVX512F() && _cd->AVX2();
```
Comment (luoyu-intel) [inline neural_speed/core/layers/mha_dense.cpp:77]: what does this mean? no avx512f but has avx2. Does it mean that MHA_AVX512F is not ready?

Comment (yuchengliu1) [reply neural_speed/core/layers/mha_dense.cpp:77]: We have not MHA_AVX512F according to @DDEle. And MHA_AVX2 is wrong on sever.

Comment (luoyu-intel) [reply neural_speed/core/layers/mha_dense.cpp:77]: Are you sure that AVX2 has no problem on client as it's wrong on server?

Comment (yuchengliu1) [reply neural_speed/core/layers/mha_dense.cpp:77]: #137 test llama on client and seems no problem.

**Comment by luoyu-intel** [comment]
@yuchengliu1 @DDEle MHA_AVX2 has a problem with phi2. It blocks the performance measurement of #174.

**Review by luoyu-intel — APPROVED** [review]

**Pull request merged**